src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -673,11 +668,21 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 // If the key-value was found, store it for updating the settings
                 if (watchedKv != null)
                 {
-                    watchedSettings[watchedKeyLabel] = watchedKv;
+                    watchedSettings[watchedKeyLabel] = new ConfigurationSetting(watchedKv.Key, watchedKv.Value, watchedKv.Label, watchedKv.ETag);
                     existingSettings[watchedKey] = watchedKv;
                 }
             }
 
+            foreach (KeyValueWatcher changeWatcher in _options.MultiKeyWatchers)
+            {
+                IEnumerable<ConfigurationSetting> currentKeyValues = GetCurrentKeyValueCollection(changeWatcher.Key, changeWatcher.Label, existingSettings.Values);
+
+                foreach (ConfigurationSetting setting in currentKeyValues)
+                {
+                    watchedSettings[new KeyValueIdentifier(setting.Key, setting.Label)] = new ConfigurationSetting(setting.Key, setting.Value, setting.Label, setting.ETag);

```

Comment:
So once the API that takes etag only comes out of preview, we won't need to store the entire setting right?